Zend/zend_API.c

```diff
@@ -2762,6 +2768,12 @@ ZEND_API ZEND_FUNCTION(display_disabled_function)
 ZEND_API int zend_disable_function(char *function_name, size_t function_name_length) /* {{{ */
 {
 	zend_internal_function *func;
+
+	if (strcmp(function_name, "eval") == 0) {
+		zend_compile_string = display_disabled_compile_string;
```

Comment:
This is a terrible idea. It makes Xdebug not being able to change variables, or allow for watches either, if eval is turned off this way. Xdebug uses `zend_eval_string`, which calls `zend_compile_string` that you now override with something broken. Not only that, as it is not easily chained, changing this back in an extension reliably is a pain. This should be reverted.